---
int8 mode
bool success
